### ISSUE TITLE
scipy.optimize.minimize fails in case of multiple constraints

### ISSUE THREAD

**Issue by pratapbhanu** [scipy/scipy#5253]
I am using 'minimize' function from scipy.optimize for an optimization problem.  **Here is the sample code to recreate the issue:**

```python
def recreate_bug():
    
    def cost(x):
        return 0.5*np.sum(x**2)
    
    A = np.ones((10,5));  b = np.ones(10)
    C = np.eye(20,5); d = np.ones(20) 
    cons =  (
                {'type':'ineq',
                 'fun': lambda x: b - np.dot(A,x)
                },
                {'type':'ineq',
                 'fun': lambda x: d - np.dot(C,x)
                }
            )
    
    minimize(fun=cost, x0=np.zeros(5), method="SLSQP", 
             constraints=cons, options={'disp':1})
```

**Here is the full "Traceback":**
```python
Traceback (most recent call last):
  File "/net/home/pratap/Documents/LiClipseWorkspace/otto/src/test_optimize.py", line 29, in <module>
    recreate_bug()
  File "/net/home/pratap/Documents/LiClipseWorkspace/otto/src/test_optimize.py", line 26, in recreate_bug
    constraints=cons, options={'disp':1})
  File "/net/home/pratap/lib/python3.4/site-packages/scipy/optimize/_minimize.py", line 435, in minimize
    constraints, callback=callback, **options)
  File "/net/home/pratap/lib/python3.4/site-packages/scipy/optimize/slsqp.py", line 400, in _minimize_slsqp
    a = concatenate((a,zeros([la,1])),1)
ValueError: all the input array dimensions except for the concatenation axis must match exactly

```

**Here is my 2 cents about the cause of this issue:**
I think the problem is due to function handle being overridden. See 'sslqp.py' file, lines 270 - 279
```python
        # check jacobian
        cjac = con.get('jac')
        if cjac is None:
            # approximate jacobian function
            def cjac(x, *args):
                return approx_jacobian(x, con['fun'], epsilon, *args)

        # update constraints' dictionary
        cons[ctype] += ({'fun': con['fun'],
                         'jac': cjac,
                         'args': con.get('args', ())}, )
```
calling the jacobian function from 'cons' dictionary for both the constraints returns the same output. Also, if you change the order of constraints in sample code, the jacobian function's output also changes but remains the same for both the constraints. The sample code works if any one of the constraints is commented out. 

**Environment Details:**
```python
SciPy version: 0.14.0
python3
OpenSUSE
```

**Comment by pv** [comment]
Try with newer scipy version, AFAIK this was fixed

**Comment by dlax** [comment]
This was fixed in 7018c110add3f52885a595da7ae1026afecb0e8f which appears in 0.14.1
